### PR TITLE
Minor cleanup of exponential backoff with jitter

### LINKED ISSUE #1720: Use exponential backoff with jitter
### What should be done?

We should use exponential backoff with jitter in places where we wait for something.

See https://aws.amazon.com/ru/blogs/architecture/exponential-backoff-and-jitter/

### Where?

Places that use `ctxutil.Sleep`, `time.Sleep`, `context.WithTimeout`, etc.

### Definition of Done

- no new external dependency;
- unit test for math added;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1720.

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/util/ctxutil/ctxutil_test.go [hunk 1]
```diff
@@ -42,4 +42,38 @@ func TestDurationWithJitter(t *testing.T) {
...
+
+		defer f.Close()
+
+		for _, task := range durations {
+			for j, duration := range task {
+				// each line has retry count (j+1) and duration waited in milliseconds.
+				fmt.Fprintln(f, j+1, duration.Milliseconds())
+			}
+		}
+	})
```
Comment (chilagrow) [inline internal/util/ctxutil/ctxutil_test.go:78]: Simple plot of output. This uses 100 tasks all retrying 5 times, yes unrealistic but it's good enough to visualised it.
The first retry is capped ~200ms (exponential part). After 4th attempt, the exponential part has reached to `cap` so it's no more exponential backoff. The scatter shows that jitter is working well.

image

**Comment by chilagrow** [comment]
@j0holo I've added minor updates to your initial work :hugs:

**Comment by j0holo** [comment]
@chilagrow Yeah, looks indeed like an nice improvement. Always nice to see different implementation of similar code.

File: internal/handlers/pg/pgdb/transactions.go [hunk 2]
```diff
@@ -109,12 +108,11 @@ func (pgPool *Pool) InTransactionRetry(ctx context.Context, f func(pgx.Tx) error
 	// TODO use exponential backoff with jitter instead
 	// https://github.com/FerretDB/FerretDB/issues/1720
```
Comment (AlekSi) [inline internal/handlers/pg/pgdb/transactions.go:109]: I think we could remove that comment now :) Please check other comments like this one

**Review by AlekSi — DISMISSED** [review]

**Review by rumyantseva — DISMISSED** [review]
Great job! 

As Alexey noticed, we need to check that the issue number `1720` is not mentioned in the code TODOs anymore. But the code itself looks reasonable and readable to me.

**Review by noisersup — DISMISSED** [review]

File: internal/handlers/pg/pgdb/transactions.go (hunk 2 above)
Comment (chilagrow) [reply internal/handlers/pg/pgdb/transactions.go:109]: removed

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by AlekSi — APPROVED** [review]

**Pull request merged**